bundle/terranova/call.go

```diff
@@ -0,0 +1,180 @@
+package terranova
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/databricks/cli/libs/dyn"
+	"github.com/databricks/cli/libs/log"
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/httpclient"
+)
+
+type HTTPClient interface {
+	MakeHTTPCall(ctx context.Context, method, path, requestBody string, response *any) error
+}
+
+type SDKHTTPClient struct {
+	Client *httpclient.ApiClient
+}
+
+func (c SDKHTTPClient) MakeHTTPCall(ctx context.Context, method, path, requestBody string, response *any) error {
+	opts := []httpclient.DoOption{httpclient.WithRequestData(requestBody)}
+	if response != nil {
+		opts = append(opts, httpclient.WithResponseUnmarshal(response))
+	}
+	return c.Client.Do(ctx, method, path, opts...)
+}
+
+type CallSpec struct {
+	Method string
+
+	// HTTP Path. Can encode resource ID as {}
+	Path string
+
+	// If present, add one pair to request dictionary: key=RequestIDField and value=ResourceID
+	RequestIDField string
+
+	// Same but convert ResourceID to integer
+	RequestIDIntegerField string
+
+	// If present, a query parameter will be added to request with key=QueryIDField and value=ResourceID
+	QueryIDField string
+
+	// If present, response will be parsed as JSON dictionary and value from key=ResponseIDField will be extracted as ResourceID
+	ResponseIDField string
+
+	// Additional processors to apply to request
+	RequestProcessors []Processor
+
+	// Additional processors to apply to response
+	ResponseProcessors []Processor
+}
+
+type Call struct {
+	Spec         *CallSpec
+	Path         string
+	RequestBody  string
+	ResponseBody any
+	StatusCode   int
+	ResponseID   string
+}
+
+func (spec *CallSpec) PrepareCall(request dyn.Value, resourceID string) (*Call, error) {
+	call := Call{
+		Spec: spec,
+	}
+
+	call.Path = spec.Path
+	if resourceID != "" {
```

Comment:
You can reverse the condition and exit earlier, can help with nesting and readability